plugins/woocommerce/src/Blocks/BlockTypes/MiniCart.php

```diff
@@ -474,15 +495,17 @@ protected function render_experimental_iapi_mini_cart( $attributes, $content, $b
 						echo $icon;
 					?>
 					<?php if ( 'never' !== $product_count_visibility ) : ?>
-						<span class="wc-block-mini-cart__badge" style="<?php echo esc_attr( $styles ); ?>"></span>
+						<span hidden data-wp-bind--hidden="!state.badgeIsVisible" data-wp-text="state.cartItemCount" class="wc-block-mini-cart__badge" style="<?php echo esc_attr( $styles ); ?>">
```

Comment:
You shouldn't need the `hidden` attribute. The iAPI's server processing should add it depending on `state.badgeIsVisible`. Same for the inner text and `state.cartItemCount`.

Reply:
Yeah, this was my bad, I forgot I could just populate `badgeIsVisible` in state. 

This should be solved now that I've refactored the cart item count to be populated in server state and I've removed the `hidden` attribute.